src/components/App.tsx

```diff
@@ -257,9 +257,9 @@ function AppRouter({ ...props }) {
   }, [state])
 
   let reset = async (identity?: any) => {
-    if (typeof identity === "undefined") {
-      sensorEventUpdate(null, LAMP.Auth._auth.id, null)
-      LAMP.SensorEvent.create(LAMP.Auth._auth.id, {
+    if (typeof identity === "undefined" && LAMP.Auth._type === "participant") {
+      await sensorEventUpdate(null, LAMP.Auth._auth.id, null)
```

Comment:
How are you `await`ing this function if it is defined as so:

```typescript
export function sensorEventUpdate(val: string, participantId: string, activityId: string, timestamp?: number) {
  ;(async () => {
    if (LAMP.Auth._type === "participant") {
      await LAMP.SensorEvent.create(participantId, {
        timestamp: timestamp ?? new Date().getTime(),
        sensor: "lamp.analytics",
        data: {
          type: "open_page",
          page: val,
          activity: activityId,
        },
      })
    }
  })()
}
```

This is literally the opposite of what you want to do and the compiler needs to be throwing an error as this is an illegal operation (you can only `await` on `async function`s). //cc @lukeoftheshire 

@ZCOEngineer Could you share more info on why this pull request was merged as you approved it?

Reply:
@avaidyam We are reviewing this, shall update here soon.

Reply:
@avaidyam This is not throwing any exceptions, as reset function is async. It worked because await returns the value of the awaited expression if it is not a promise. But there was one issue with retrieving participant id which was the cause of {
  "error": "403.security-context-out-of-scope"
}
I have also updated SensorEventUpdate function to async function.
https://github.com/BIDMCDigitalPsychiatry/LAMP-dashboard/pull/549

Reply:
> It worked because await returns the value of the awaited expression if it is not a promise.

That's correct but also the wrong approach, as the non-`async` function we're `await`ing actually wraps `async` code. Thanks for fixing this in #549.